26: Add topic page

EugeneTurkin opened the pull request:
After we've added topics creation (#25) and ability to see list of created topics (#23), we can go further and add a page for topic itself.

In the scope of this task we need to add an ability to see topic content at `/topics/{topic_id}`. For now it will only be author, title, description and creation date (no posts listed for now)

Steps to do:
- add a template for topic page
- add a router for `/topics/{topic_id}` which will get topic from database and return correct template rendered
- add an ability to go to the appropriate topic page by the link for each topic in `/topics`

birthdaysgift commented on src/routes.py:
```diff
@@ -88,15 +88,34 @@ def topics() -> str | Response:
+
+
+@bp.route("/topics/<topic_id>")
+def topic_page(topic_id: str) -> str | Response:
```
Seems like `topic_id` should be `int`, not `str`. I think flask has an ability to get route parameter as integer. We need to try it.

EugeneTurkin committed:
26: Add topic page

After we've added topics creation (#25) and ability to see list of created topics (#23), we can go further and add a page for topic itself.

In the scope of this task we need to add an ability to see topic content at `/topics/{topic_id}`. For now it will only be author, title, description and creation date (no posts listed for now)

Steps to do:

    - add a template for topic page
    - add a router for `/topics/{topic_id}` which will get topic from database and return correct template rendered
    - add an ability to go to the appropriate topic page by the link for each topic in `/topics`

birthdaysgift commented on src/routes.py:
```diff
@@ -105,3 +105,17 @@ def create_topic() -> str | Response:
     return redirect(url_for("routes.login"))
+
+
+@bp.route("/topics/<topic_id>")
```
I'm not sure that providing a mypy type in the signature is enough. If it works now it's a little bit weird and probably needs a little research.

And why I'm telling that is because I expected not only the changes in mypy type but in the flask route as well.
I expected it to look like in [this](https://flask.palletsprojects.com/en/1.1.x/quickstart/#variable-rules) article.
